sys/log/full/syscfg.yml

```diff
@@ -115,6 +115,25 @@ syscfg.defs:
         value: 0
         restrictions: LOG_STORAGE_INFO
 
+    LOG_PERSIST_WATERMARK:
+        description: Enable persisted storage of watermarks.
+        value: 1
+
+    LOG_REBOOT_INIT_WATERMARK_UNK:
+        description: >
+            Initialize watermarks to LOG_INIT_WATERMARK_VAL on reboots to
+            indicate unknown watermarks
+        value: 0
+        restrictions:
+            - LOG_PERSIST_WATERMARK == 0
+            - LOG_STORAGE_WATERMARK
+
+    LOG_INIT_WATERMARK_VAL:
+        description: >
+            Value to initialize watermarks to
+            to indicate unknown watermarks
+        value: 0xFFFFFFFF
+
```

Comment:
Personally, I think this is too much configurability.  The ability to configure behavior is good, but it comes at a cost: extra complexity.  If I can't think of a case where someone would ever want to override a setting, I think it is best to remove the setting.

* `LOG_PERSIST_WATERMARK`: Makes sense.
* `LOG_REBOOT_INIT_WATERMARK_UNK`: I think this naturally follows from the persist setting.  I just can't think of a case where someone would want a) persist=1 && unk=1, or b) persist=0 && unk=0.
* `LOG_INIT_WATERMARK_VAL`: Would someone really ever want to use a value other than 0xffffffff here?

So my suggestion is to remove the second two settings.  If persistence is disabled, always initialize the watermark to 0xffffffff.